lua/gitsigns/manager.lua

```diff
@@ -26,6 +26,52 @@ local signs_staged --- @type Gitsigns.Signs
 
 local M = {}
 
+local statuscolumn_active = false
+
+--- @param bufnr? integer
+--- @param top? integer
+--- @param bot? integer
+local function redraw_statuscol(bufnr, top, bot)
+  if statuscolumn_active then
+    api.nvim__redraw({
+      buf = bufnr,
+      range = { top, bot },
+      statuscolumn = true,
+    })
+  end
+end
+
+function M.statuscolumn()
+  if not statuscolumn_active then
+    config.signcolumn = false
+    statuscolumn_active = true
+  end
+
+  local res = {}
+  local res_len = 0
+  local max_pad = 0
+  local lnum = vim.v.lnum
+  for _, signs in pairs({ signs_normal, signs_staged }) do
+    if next(signs.signs) then
+      max_pad = 2
+    end
+    local marks = api.nvim_buf_get_extmarks(0, signs.ns, { lnum - 1, 0 }, { lnum - 1, -1 }, {})
```

Comment:
```suggestion
		local buf = api.nvim_win_get_buf(vim.g.statusline_winid)
    local marks = api.nvim_buf_get_extmarks(, signs.ns, { lnum - 1, 0 }, { lnum - 1, -1 }, {})
```

I suppose there is no need to change the function signature like I suggested on matrix.

Reply:
`vim.g.statusline_winid` doesn't seem to be set when `statuscolumn` runs.